### ISSUE TITLE
Problem with Eventchannel when an event field has the character " followed by , or the word 'Event' at the message field

### ISSUE THREAD

**Issue by cristgl** [wazuh/wazuh#3343]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 3.9.0-3919 | Logcollector-EventChannel | Manager/Agent | Sources | Linux/Windows |

The agent sends every event to the manager with the next format: `{"Message": "message", "Event": "<Event></Event>"}`.

It is an XML inside a JSON object, which can be extended to have some more JSON fields in the future. The [code](https://github.com/wazuh/wazuh/blob/3b2961211f9af83d914471e8380c1334756f39e3/src/analysisd/decoders/winevtchannel.c#L117) finds the word `Event` and tries to get the entire XML by searching for the next `"` followed by a `,` (just in case there are some more fields inside the JSON object) or a `}`. This can fail for some events that have these two characters together, for example:

```
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
- <System>
  <Provider Name="Microsoft-Windows-FilterManager" Guid="{F3C5E28E-63F6-49C7-A204-E48A1BC4B09D}" /> 
  <EventID>6</EventID> 
  <Version>1</Version> 
  <Level>4</Level> 
  <Task>0</Task> 
  <Opcode>0</Opcode> 
  <Keywords>0x8000400000000000</Keywords> 
  <TimeCreated SystemTime="2019-05-17T13:25:44.781843600Z" /> 
  <EventRecordID>250</EventRecordID> 
  <Correlation /> 
  <Execution ProcessID="4" ThreadID="2712" /> 
  <Channel>System</Channel> 
  <Computer>DESKTOP-UGQJNVU</Computer> 
  <Security UserID="S-1-5-18" /> 
  </System>
- <EventData>
  <Data Name="FinalStatus">0x0</Data> 
  <Data Name="DeviceVersionMajor">6</Data> 
  <Data Name="DeviceVersionMinor">2</Data> 
  <Data Name="DeviceNameLength">4</Data> 
  <Data Name="DeviceName">KLIF</Data> 
  <Data Name="DeviceTime">2018-09-28T04:25:08.000000000Z</Data> 
  <Data Name="ExtraInfoLength">202</Data> 
  <Data Name="ExtraInfoString">{ "flags" : "0x00000018" , "registration_version" : "0x00000203" , "tx" : false , "sections" : true , "frame" : 0 , "class_name" : "FSFilter Activity Monitor" , "instances" : [["320400","0x00000000"]] }</Data> 
  <Data Name="FilterID">{02000000-000B-0000-0CD4-2608B40CD501}</Data> 
  </EventData>
  </Event>
```

This means that this event is not taken into account, the next warning is shown:
```
WARNING: Could not read XML string: '<Event xmlns='http://schemas.microsoft.com/win/2004/08/events/event'><System><Provider Name='Microsoft-Windows-FilterManager' Guid='{F3C5E28E-63F6-49C7-A204-E48A1BC4B09D}'/><EventID>6</EventID><Version>1</Version><Level>4</Level><Task>0</Task><Opcode>0</Opcode><Keywords>0x8000400000000000</Keywords><TimeCreated SystemTime='2019-05-17T13:25:44.781843600Z'/><EventRecordID>250</EventRecordID><Correlation/><Execution ProcessID='4' ThreadID='2712'/><Channel>System</Channel><Computer>DESKTOP-UGQJNVU</Computer><Security UserID='S-1-5-18'/></System><EventData><Data Name='FinalStatus'>0x0</Data><Data Name='DeviceVersionMajor'>6</Data><Data Name='DeviceVersionMinor'>2</Data><Data Name='DeviceNameLength'>4</Data><Data Name='DeviceName'>KLIF</Data><Data Name='DeviceTime'>2018-09-28T04:25:08.000000000Z</Data><Data Name='ExtraInfoLength'>202</Data><Data Name='ExtraInfoString'>{ \"flags\" : \"0x00000018\" , \"registration_version\" : \"0x00000203\" , \"tx\" : false , \"sections\" : true , \"frame\" : 0 , \"class_name\" : \"FSFilter Activity Monitor\" , \"instances\" : [[\"320400\'
```

Same issue happening with the 'Event' word at the message field. Regarding the [code](https://github.com/wazuh/wazuh/blob/3b2961211f9af83d914471e8380c1334756f39e3/src/analysisd/decoders/winevtchannel.c#L115), the event part that comes after it cannot be parsed correctly. For example, the next event sent by the agent:

```
{
   "Message":"Event type:     Application settings changed Application\\\\Name:     Kaspersky Endpoint Security for Windows User:     DESKTOP-UGQJNVU\\\\helo (Active user) Component:     Protection ",
   "Event":"<Event xmlns='http://schemas.microsoft.com/win/2004/08/events/event'><System><Provider Name='avp'/><EventID Qualifiers='16386'>218</EventID><Level>4</Level><Task>0</Task><Keywords>0x8</Keywords><TimeCreated SystemTime='2019-05-17T14:38:39.235350400Z'/><EventRecordID>42</EventRecordID><Channel>Kaspersky Event Log</Channel><Computer>DESKTOP-UGQJNVU</Computer><Security/></System><EventData><Data>Event type:     Application settings changed Application\\\\Name:     Kaspersky Endpoint Security for Windows User:     DESKTOP-UGQJNVU\\\\helo (Active user) Component:     Protection </Data></EventData></Event>"
}
```

Our event would start right after the word `Event`, this does not retrieve a valid XML and this is why the event cannot be parsed. At the end, the information we get at the archives.log file is:
```
2019 May 17 15:28:16 (w10) any->EventChannel {"win":{"system":{"message":"Event type:     Product started Application\\Name:     Kaspersky Endpoint Security for Windows User:     DESKTOP-UGQJNVU\\helo (Active user) Component:     Protection"}}}
```

Without any information about the event.

To develop a solution, the changes made in this [PR](https://github.com/wazuh/wazuh/pull/3320) should be taken into account, as the new format that the agent sends the message with is: `{"Event":"<Event></Event>} Message: message`.

**Comment by cristgl** [comment]
The solution for this issue is being develop taking into account this one: https://github.com/wazuh/wazuh/issues/3242. Some comments about the development are being added to the mentioned issue's PR: https://github.com/wazuh/wazuh/pull/3320

**Comment by chemamartinez** [comment]
Fixed at #3322
